packages/transport/src/modules/tokens/TokenController.ts

```diff
@@ -187,11 +190,16 @@ export class TokenController extends TransportController {
     }
 
     public async loadPeerTokenByReference(tokenReference: TokenReference, ephemeral: boolean): Promise<Token> {
+        // TODO: add the token reference forIdentity once available
         return await this.loadPeerToken(tokenReference.id, tokenReference.key, ephemeral);
     }
 
-    public async loadPeerToken(id: CoreId, secretKey: CryptoSecretKey, ephemeral: boolean): Promise<Token> {
+    public async loadPeerToken(id: CoreId, secretKey: CryptoSecretKey, ephemeral: boolean, forIdentity?: CoreAddress): Promise<Token> {
         const tokenDoc = await this.tokens.read(id.toString());
+        if (!tokenDoc && forIdentity && !forIdentity.equals(this.parent.identity.address)) {
+            // if you created the token, it exists already
+            throw TransportCoreErrors.general.notIntendedForYou(id.toString());
+        }
```

Comment:
you can not throw here. you have to do it after the call to `await this.updateCacheOfToken(token);` below. PLease also check why the test didn't shows this.

Reply:
I don't understand the issue, I'm not using data from the existing token

Reply:
Why should we not throw here? If it is not intended for you, there should be nothing in the DB and the update of cache would throw an 404.

Reply:
How will it come to the db?

Reply:
Should it go to the DB if it is not intended for you ?